### PR TITLE
AMDGPU: Support local atomicrmw fmin/fmax for float/double

### PR DESCRIPTION
**Pull request by arsenm**
This has always been supported. Somehow, we ended up with 2
copies of clang builtins for this case, and the newer one
erroneously requires gfx8-insts.

### CODE REVIEW

**Comment by arsenm** [comment]
* **#95592** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95592?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#95591** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95591?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#95590** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/95590?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp [hunk 1]
```diff
@@ -16042,7 +16027,15 @@ SITargetLowering::shouldExpandAtomicRMWInIR(AtomicRMWInst *RMW) const {
     return AtomicExpansionKind::CmpXChg;
   }
   case AtomicRMWInst::FMin:
-  case AtomicRMWInst::FMax:
+  case AtomicRMWInst::FMax: {
+    Type *Ty = RMW->getType();
+
+    // LDS float and double fmin/fmax were always supported.
+    if (AS == AMDGPUAS::LOCAL_ADDRESS && (Ty->isFloatTy() || Ty->isDoubleTy()))
+      return AtomicExpansionKind::None;
```
Comment (rampitec) [inline llvm/lib/Target/AMDGPU/SIISelLowering.cpp:16035]: Can it be unsafe? I.e. with denorm handling?

Also 4 cases below now all integer but checking isFloatTy().

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:16035]: No, LDS atomics are always well behaved. I think denormals always worked correctly even for the global cases too for fmin/fmax

Comment (rampitec) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:16035]: OK. What's with isFloatTy() below?

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:16035]: Should be dead code, but in principle the unsafe handling would be handling the no-fine-grained-remote-memory case, not denormal mishandling

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:16035]: > would be handling the no-fine-grained-remote-memory case, not denormal mishandling

Which don't matter for LDS

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:16035]: The isFloatTy is not dead code, but it triggers a behavior change in umin/umax/smin/smax, so I'll fix this separately

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp:16035]: > The isFloatTy is not dead code, but it triggers a behavior change in umin/umax/smin/smax, so I'll fix this separately

Nevermind, this was only due to interpreting unsafeFP attribute for the integer case

**Review by rampitec — APPROVED** [review]
LGTM

**Pull request merged**